prow/plugins/owners-label/owners-label.go

```diff
@@ -84,10 +99,20 @@ func handle(ghc githubClient, oc ownersClient, log *logrus.Entry, pre *github.Pu
 		neededLabels.Insert(oc.FindLabelsForFile(change.Filename).List()...)
 	}
 
-	for _, toAdd := range neededLabels.Difference(currentLabels).List() {
-		if err := ghc.AddLabel(pre.Repo.Owner.Login, pre.Repo.Name, pre.Number, toAdd); err != nil {
-			log.WithError(err).Errorf("Adding %q label.", toAdd)
+	var nonexistent []string
+
+	for _, labelToAdd := range neededLabels.Difference(currentLabels).List() {
+		if _, ok := existingLabels[labelToAdd]; !ok {
```

Comment:
`labelToAdd` has may contain capital letters so `existingLabels` should be keyed by the raw existing label values, not  `strings.ToLower(l.Name)`.
This also allows `existingLabels` to become a set instead of a map.